internal/controllers/import_controller.go

```diff
@@ -303,27 +306,21 @@ func (r *CAPIImportReconciler) shouldAutoImport(ctx context.Context, capiCluster
 	return autoImport, nil
 }
 
-func (r *CAPIImportReconciler) reconcileDelete(ctx context.Context, capiCluster *clusterv1.Cluster,
-	rancherCluster *rancher.Cluster,
-) (ctrl.Result, error) {
+func (r *CAPIImportReconciler) reconcileDelete(ctx context.Context, capiCluster *clusterv1.Cluster) (ctrl.Result, error) {
 	log := log.FromContext(ctx)
-	log.Info("Reconciling CAPI Cluster Delete")
+	log.Info("Reconciling rancher cluster deletion")
 
-	// If the Rancher Cluster has ClusterImportedAnnotation annotation that said we imported it,
-	// then annotate the CAPI cluster so that we don't auto-import again.
-	if turtlesannotations.HasClusterImportAnnotation(rancherCluster) {
-		log.Info(fmt.Sprintf("rancher cluster %s has %s annotation, so annotating CAPI cluster %s",
-			rancherCluster.Name, turtlesannotations.ClusterImportedAnnotation, capiCluster.Name))
-
-		annotations := capiCluster.GetAnnotations()
-		if annotations == nil {
-			annotations = map[string]string{}
-		}
+	// If the Rancher Cluster was already imported, then annotate the CAPI cluster so that we don't auto-import again.
```

Comment:
where is the check in the code from this comment, I seem to be missing it

Reply:
https://github.com/rancher-sandbox/rancher-turtles/pull/87/files#diff-df81a5620bfede50d1c70fd0ebb982e652d28721e15b825c52dddcf97ceda0deR384

Reply:
can we log the message in that case, it was not obvious from first look

Reply:
Operator does not reconcile non-imported rancher clusters. Imported rancher cluster follows the `<capi-cluster>-capi` naming convention. Therefore it is not possible here that any of the reconcile code, including reconcile delete would be executed on non-imported rancher cluster, unless it follows given naming convention. So log message on the  https://github.com/rancher-sandbox/rancher-turtles/pull/87/files#diff-df81a5620bfede50d1c70fd0ebb982e652d28721e15b825c52dddcf97ceda0deR311 and https://github.com/rancher-sandbox/rancher-turtles/pull/87/files#diff-df81a5620bfede50d1c70fd0ebb982e652d28721e15b825c52dddcf97ceda0deR314 are both describing what is happening with the cluster. Don’t see a need to make it more vivid than that.

Reply:
It will also be logged in the predicate itself. My concern is that as far as I know predicates might not always work as expected and we should structure our code like they don't exist, it's nice if they filter out resources for us but don't expect them to always do that. I wouldn't remove the code.